s2s/time.py

```diff
@@ -414,41 +412,108 @@ def __str__(self):
         return f"{self._n_intervals} periods of {self.freq} leading up to {self.month}/{self.day}."
 
     def __repr__(self):
-        props = ", ".join([f"{k}={v}" for k, v in self.__dict__.items() if not k.startswith('_')])
+        if self._intervals is not None:
+            return repr(self._intervals)
+
+        props = ", ".join(
+            [f"{k}={v}" for k, v in self.__dict__.items() if not k.startswith("_")]
+        )
+        return f"AdventCalendar({props})"
+
+    def _repr_html_(self):
+        """For jupyter notebook to load html compatiable version of __repr__."""
+        if self._intervals is not None:
+            return self._intervals._repr_html_() # pylint: disable=protected-access
+
+        props = ", ".join(
+            [f"{k}={v}" for k, v in self.__dict__.items() if not k.startswith("_")]
+        )
         return f"AdventCalendar({props})"
 
-    def discard(self, max_lag):
+    @property
+    def flat(self):
+        if self._intervals is not None:
+            return self._intervals.stack()
+        raise ValueError("The calendar is not initialized with intervals yet.")
```

Comment:
Still to do: clarify this error/convert to warning by saying something like: "use calendar.map_years or calendar.map_to_data"

Reply:
"use calendar.map_years or calendar.map_to_data" is added to the error message now.